src/Microsoft.SqlTools.ServiceLayer/Metadata/SmoScripterHelpers.cs

```diff
@@ -172,12 +172,17 @@ private static IEnumerable<string> GenerateTableScripts(Server server)
             return scripts;
         }
 
-        private static UrnCollection GetAllServerTableAndViewUrns(Server server)
+        private static UrnCollection GetDatabaseTableAndViewUrns(Server server, string databaseName)
         {
             UrnCollection urnCollection = new UrnCollection();
 
             foreach (Database db in server.Databases)
             {
+                if (db.Name != databaseName)
```

Comment:
Instead of looping through all databases you can just access the DB directly through the indexer on Databases